### PR TITLE
Leaderboard

### PR DESCRIPTION
**Pull request by wekul**
Fixed the mass of issues i created for myself. Added leaderboard command with ability to filter between all time, past 30 and 7 days. All enclosed in a nice view but still i do not know if it works 😎. The code to start migrating it to a PIL render is there just good luck

### CODE REVIEW

File: cogs/stats/leaderboard.py [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+    BG_COLOR = discord.Color.from_str("#1b1d21")
+
+    """WIP. Using embed for now"""
+
+
+class LeaderboardView(discord.ui.View):
+    def __init__(self, embed: LeaderboardEmbed, author: discord.User | discord.Member):
+        self.author = author
+        self.current_embed: LeaderboardEmbed = embed
+        super().__init__(timeout=300)
```
Comment (LeoCx1000) [inline cogs/stats/leaderboard.py]: Here you should subclass `utils.View` instead, so errors properly log to the webhook, and take a parameter of `bot` which gets passed to super init.

File: cogs/stats/leaderboard.py [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+        ORDER BY message_count DESC LIMIT 10
+        """
+        self._data: list[asyncpg.Record] = await self._pool.fetch(
+            query.format("--" if interval is None else f"AND created_at > NOW - INTERVAL {interval}"), False
+        )
+
+        if not self._data:
+            raise RuntimeError("No leaderboard can be generated.")
+
+        for rank, user in enumerate(self._data):
```
Comment (LeoCx1000) [inline cogs/stats/leaderboard.py]: Shouldn't be 0-indexed
```suggestion
        for rank, user in enumerate(self._data, start=1):
```

**Review by LeoCx1000 — CHANGES_REQUESTED** [review]
A few things I noticed

**Comment by wekul** [comment]
Changes have been added, along with the editing of LeaderboardEmbed to allow for bot to be taken on startup

**Comment by LeoCx1000** [comment]
Nice. One last thing, make it display the amount of messages sent, using `:,` to format them with commas

```python
f"... {amount:,}"
```

**Comment by wekul** [comment]
KABOOM

**Review by LeoCx1000 — APPROVED** [review]
All seems fine now. Thanks, and can't wait to see the PIL one but I need to figure out a better way to local test before that happens.

**Pull request merged**